### PR TITLE
Add job progression dates

### PR DESCRIPTION
**Pull request by nathancashmoredwp**

### CODE REVIEW

File: app/controllers/jobs-controller.js [hunk 1]
```diff
@@ -34,6 +39,9 @@ router.post('/new', (req, res, next) => {
   const jobData = Object.assign({}, req.body, {
     accountId,
     status_sort_index: progression.getById(req.body.status).order,
+    deadlineDate: parseDateOrReturnNull(req.body.deadlineDate),
+    applicationDate: parseDateOrReturnNull(req.body.applicationDate),
+    interviewDate: parseDateOrReturnNull(req.body.interviewDate),
```
Comment (lukasz-kaniowski) [inline app/controllers/jobs-controller.js]: only store in db date that corresponds to selected status

File: test/features/support/fixtures.js [hunk 2]
```diff
@@ -6,5 +6,8 @@ module.exports = {
     sourceUrl: 'http://example.com',
     rating: '4',
     deadline: '20/10/2016',
+    deadlineDate: null,
+    applicationDate: null,
+    interviewDate: null,
```
Comment (lukasz-kaniowski) [inline test/features/support/fixtures.js]: maybe it is worth adding a missing `status` field?
date for status should not be null

File: test/integration/support/db-helper.js [hunk 3]
```diff
@@ -13,6 +13,9 @@ const sampleJob = {
   rating: '1',
   status: 'applied',
   status_sort_index: 1,
+  deadlineDate: null,
+  applicationDate: null,
```
Comment (lukasz-kaniowski) [inline test/integration/support/db-helper.js]: date for status should not be null

File: app/models/progression.js [hunk 4]
```diff
@@ -20,6 +20,10 @@ class Progression {
   getInitialSubset() {
     return this.model.slice(0, 3).map(item => item.id);
   }
+
+  getDateField(id) {
+    return this.model.find(item => item.id === id).dateField;
+  }
```
Comment (lukasz-kaniowski) [inline app/models/progression.js]: how are you going to handle `null` scenario? maybe it is worth adding `hasDateField` method?

File: app/controllers/jobs-controller.js (hunk 1 above)
Comment (nathancashmoredwp) [inline app/controllers/jobs-controller.js]: Intend to only store dates that have been entered but if i'm adding a job and i'm at the application stage i may well input what the deadline date was as well as the application date and i'm not sure we should restrict that.

File: test/features/support/fixtures.js (hunk 2 above)
Comment (nathancashmoredwp) [inline test/features/support/fixtures.js]: Have changed this to be more relevant for the status selected e.g. added status = applied and a applicationDate.  Removed the nulls.

File: test/integration/support/db-helper.js (hunk 3 above)
Comment (nathancashmoredwp) [inline test/integration/support/db-helper.js]: as above... Have changed this to be more relevant for the status selected e.g. added status = applied and a applicationDate. Removed the nulls.

File: app/models/progression.js (hunk 4 above)
Comment (nathancashmoredwp) [inline app/models/progression.js]: Added hasDateField and incorporated into view which should make update job and any additional date fields easier to accomodate

**Comment by lukasz-kaniowski** [comment]
Would be good to:
* persist dates on the form when there is validation error
* only save selected date to database if there are multiple send in the post

**Pull request merged**